src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -47,24 +47,36 @@ static int osrandom_rand_status(void) {
     }
 }
 #else
-static int urandom_fd = -1;
+static struct {
+    int fd;
+    dev_t st_dev;
+    ino_t st_ino;
+} urandom_cache = { -1 };
 
 static int osrandom_finish(ENGINE *e);
 
 static int osrandom_init(ENGINE *e) {
-    if (urandom_fd > -1) {
+    struct stat st;
+
+    if (urandom_cache.fd > -1) {
         return 1;
     }
-    urandom_fd = open("/dev/urandom", O_RDONLY);
-    if (urandom_fd > -1) {
-        int flags = fcntl(urandom_fd, F_GETFD);
+    urandom_cache.fd = open("/dev/urandom", O_RDONLY);
+    if (urandom_cache.fd > -1) {
+        int flags = fcntl(urandom_cache.fd, F_GETFD);
         if (flags == -1) {
             osrandom_finish(e);
             return 0;
-        } else if (fcntl(urandom_fd, F_SETFD, flags | FD_CLOEXEC) == -1) {
+        } else if (fcntl(urandom_cache.fd, F_SETFD, flags | FD_CLOEXEC) == -1) {
             osrandom_finish(e);
             return 0;
         }
+        if (fstat(urandom_cache.fd, &st)) {
```

Comment:
So `fstat` should return -1 (on error) or 0. Do we want to check for -1 explicitly? I know it's "truthy" but, it might be better to be explicit (like we are below on L90).

Also, do we always expect `orandom_finish` to be the "right thing" here? Should we be checking errno just to be safe?